modules/siddhi-core/src/main/java/io/siddhi/core/util/cache/CacheUtils.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.siddhi.core.util.cache;
+
+import io.siddhi.core.event.state.StateEvent;
+import io.siddhi.core.event.stream.StreamEvent;
+import io.siddhi.core.executor.ConstantExpressionExecutor;
+import io.siddhi.core.executor.ExpressionExecutor;
+import io.siddhi.core.executor.VariableExpressionExecutor;
+import io.siddhi.core.util.collection.executor.AndMultiPrimaryKeyCollectionExecutor;
+import io.siddhi.core.util.collection.executor.CompareCollectionExecutor;
+import io.siddhi.core.util.collection.operator.CompiledCondition;
+import io.siddhi.core.util.collection.operator.OverwriteTableIndexOperator;
+
+import java.util.List;
+
+/**
+ * class containing utils related to store table cache
+ */
+public class CacheUtils {
+    public static int findEventChunkSize(StreamEvent streamEvent) {
+        int chunkSize = 1;
+        if (streamEvent == null) {
+            return 0;
+        }
+        StreamEvent streamEventCopy = streamEvent;
+        while (streamEventCopy.hasNext()) {
+            chunkSize = chunkSize + 1;
+            streamEventCopy = streamEventCopy.getNext();
+        }
+        return chunkSize;
+    }
+
+    public static String getPrimaryKey(CompiledCondition compiledCondition, StateEvent matchingEvent) {
```

Comment:
Where is this used ?

Reply:
Let's name it as `genrateCompositePrimaryKey`

Reply:
it was used previously when I recursively went through compiled condition when event comes.now i check the condition when cc is created so these functions are not needed. will remove